### DISCUSSION TITLE
Subscribe to array/item in array?

### DISCUSSION THREAD

**Discussion by piscopancer** [pmndrs/valtio#849] [Q&A] [answered]
Let's say I have a component that should track the name in the store it was given in props
```tsx
const namesStore = proxy({
  names: [] as string[],
})

function NameFinderComponent({name} : {name: string}) {
  const nameSnap = useSnapshot(namesStore.names.find((s) => s === name) /** can be undefined */)

  return nameSnap ? 'such name exists' : null
}
```
However the name can be not found, leading to inability to subscribe to it. I would probably have to subscribe to array then, right?

**Comment by dai-shi** [comment, marked as answer]
> I would probably have to subscribe to array then, right?

Yes.
